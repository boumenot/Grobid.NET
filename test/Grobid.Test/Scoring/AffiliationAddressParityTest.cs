using System.Linq;
using System.Reflection;
using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using Grobid.NET.Scoring;
using Xunit;

namespace Grobid.Test.Scoring
{
    // Evaluate the parity between Grobid.NET and Grobid for the Affiliation-Address model.
    [UseReporter(typeof(BeyondCompare4Reporter))]
    public class AffiliationAddressParityTest 
    {
        [Fact]
        public void AffiliationAddressModelComparisonAgainstGrobid()
        {
            var features = Assembly.GetExecutingAssembly().GetManifestResourceStream("Grobid.Test.content.official.affiliation-address.labeled.txt")
                .ToLines()
                .Where(x => !string.IsNullOrWhiteSpace(x))
                .Select(x => new
                {
                    Line = x,
                    Match = DateParityTest.LabeledRx.Match(x)
                })
                .Select(x => new
                {
                    Expected = x.Match.Groups[1].Value,
                    Actual = x.Match.Groups[2].Value,
                    x.Line,
                })
                .ToArray();

            var sb = new StringBuilder();
            var scorer = new ModelScorer();
            foreach (var x in features)
            {
                scorer.Eval(x.Expected, x.Actual);
            }

            sb.AppendLine($"==== Token Level Result ====\n");
            var modelScoreCard = new ModelScoreCard();
            modelScoreCard.Render(scorer.Stat, sb);

            var s = sb.ToString();
            Approvals.Verify(sb);
        }

    }
}
