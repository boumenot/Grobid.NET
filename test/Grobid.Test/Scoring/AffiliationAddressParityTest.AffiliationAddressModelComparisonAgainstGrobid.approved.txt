==== Token Level Result ====

           Label   Accuracy  Precision     Recall         F0
        addrLine     0.9989     0.9881     1.0000     0.9940
         country     0.9997     1.0000     0.9952     0.9976
      department     0.9954     0.9890     0.9922     0.9906
     institution     0.9976     0.9910     0.9955     0.9932
      laboratory     0.9981     1.0000     0.9551     0.9770
          marker     1.0000     1.0000     1.0000     1.0000
           other     0.9987     0.9969     0.9954     0.9961
         postBox     0.9995     1.0000     0.9697     0.9846
        postCode     1.0000     1.0000     1.0000     1.0000
          region     0.9989     0.9783     0.9375     0.9574
      settlement     0.9987     0.9828     0.9957     0.9892

      MICRO AVG.     0.9987     0.9918     0.9921     0.9920
      MACRO AVG.     0.9987     0.9929     0.9841     0.9884
